### ISSUE TITLE
ArgumentNullException "Value cannot be null. (Parameter 'values')" at 7-1-2024 23:14:28 +01:00

### ISSUE THREAD

**Issue by Havym** [njthomson/SrvSurvey#37]
### What happened?

Scanning planet while in DSS.

### Version

1.1.0.4

### Exception message

Value cannot be null. (Parameter 'values')

### Exception stack

   at System.String.JoinCore[T](ReadOnlySpan`1 separator, IEnumerable`1 values)
   at System.String.Join[T](String separator, IEnumerable`1 values)
   at SrvSurvey.game.Game.initCargoData() in D:\a\SrvSurvey\SrvSurvey\SrvSurvey\game\Game.cs:line 669
   at SrvSurvey.game.Game.initializeFromJournal(LoadGame loadEntry) in D:\a\SrvSurvey\SrvSurvey\SrvSurvey\game\Game.cs:line 486
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
   at SrvSurvey.game.Game.Journals_onJournalEntry(JournalEntry entry, Int32 index) in D:\a\SrvSurvey\SrvSurvey\SrvSurvey\game\Game.cs:line 708

### Relevant log output

```
23:04:32: SrvSurvey version: 1.1.0.4
23:04:32: New log file: C:\Users\User\AppData\Roaming\SrvSurvey\SrvSurvey\1.1.0.0\logs\srvs-20240107_230432.txt
23:04:32: dataFolder: C:\Users\User\AppData\Roaming\SrvSurvey\SrvSurvey\1.1.0.0
23:04:32: Removing old log file: C:\Users\User\AppData\Roaming\SrvSurvey\SrvSurvey\1.1.0.0\logs\srvs-20240106_154029.txt
23:04:32: Loaded settings: {
  "preferredCommander": "",
  "lastCommander": "Havym",
  "lastFid": "F5439629",
  "watchedJournalFolder": "C:\\Users\\User\\Saved Games\\Frontier Developments\\Elite Dangerous\\",
  "hideJournalWriteTimer": true,
  "targetLatLong": {
    "lat": 0.0,
    "long": 0.0
  },
  "targetLatLongActive": false,
  "autoShowBioSummary": true,
  "autoShowBioPlot": true,
  "autoShowPlotFSS": true,
  "autoShowGuardianSummary": true,
  "autoShowRamTah": true,
  "autoShowPlotSysStatus": true,
  "skipGasGiantDSS": true,
  "skipRingsDSS": false,
  "skipLowValueDSS": true,
  "skipLowValueAmount": 1000000,
  "autoTrackCompBioScans": true,
  "skipAnalyzedCompBioScans": true,
  "autoRemoveTrackerOnSampling": true,
  "useExternalData": true,
  "autoLoadPriorScans": true,
  "skipPriorScansLowValue": true,
  "skipPriorScansLowValueAmount": 1000000,
  "showCanonnSignalsOnRadar": true,
  "useSmallCirclesWithCanonn": true,
  "hideMyOwnCanonnSignals": true,
  "focusGameOnStart": true,
  "focusGameOnMinimize": true,
  "enableGuardianSites": true,
  "enableEarlyGuardianStructures": false,
  "disableRuinsMeasurementGrid": false,
  "disableAerialAlignmentGrid": false,
  "hidePlottersFromCombatSuits": true,
  "hideOverlaysFromMouse": false,
  "autoShowFlightWarnings": true,
  "highGravityWarningLevel": 1.0,
  "plotterOpacity": 50.0,
  "formMainLocation": "876, 96",
  "formLogsLocation": "0, 0, 0, 0",
  "formAllRuinsLocation": "0, 0, 0, 0",
  "formRuinsLocation": "0, 0, 0, 0",
  "formBeaconsLocation": "0, 0, 0, 0",
  "formMapEditor": "0, 0, 0, 0",
  "formRamTah": "0, 0, 0, 0",
  "mapShowNotes": true,
  "blinkTigger": 134217728,
  "blinkDuration": 3000,
  "processScreenshots": false,
  "addBannerToScreenshots": true,
  "deleteScreenshotOriginal": false,
  "useGuardianAerialScreenshotsFolder": true,
  "screenshotSourceFolder": "C:\\Users\\User\\Pictures\\Frontier Developments\\Elite Dangerous",
  "screenshotTargetFolder": "C:\\Users\\User\\Pictures\\Frontier Developments\\Elite Dangerous\\converted",
  "rotateAndTruncateAlphaAerialScreenshots": true,
  "aerialAltAlpha": 1200.0,
  "aerialAltBeta": 1550.0,
  "aerialAltGamma": 1600.0,
  "idxGuardianPlotter": 0,
  "migratedAlphaSiteHeading": false,
  "inferColor": "102, 255, 255",
  "inferTolerance": 25,
  "inferThreshold": 0.002,
  "dataFolder1100": true,
  "pubDataSettlementTemplate": 9,
  "pubDataGuardian": 9,
  "processIdx": 0
}
23:04:32: Watching folder: C:\Users\User\Saved Games\Frontier Developments\Elite Dangerous\
23:04:32: Watching file: C:\Users\User\AppData\Local\Frontier Developments\Elite Dangerous\Options\Graphics\DisplaySettings.xml
23:04:32: setFocusED: got Zero!
23:04:32: updatePubData ...
23:04:32: isMigrationValid: False, dataFolder1100: True
23:04:33: Loaded 512 ruins, 30 beacons, 155 beacons
23:04:33: Using settlementTemplates.json from pubData
23:04:33: Reading settlementTemplates.json: C:\Users\User\AppData\Roaming\SrvSurvey\SrvSurvey\1.1.0.0\pub\settlementTemplates.json
23:04:33: SiteTemplate.Imported 13 templates
23:04:33: CodexRef init ...
23:04:33: loadCodexRef: reading codex/ref from disk
23:04:33: prepBioRef: reading bioRef from disk
23:04:33: loadOrganicRewards: reading organic rewards from disk
23:04:33: CodexRef init - complete
23:04:33: pubDataSettlementTemplate - local: 9, remote: 9
23:04:33: pubDataGuardian - local: 9, remote: 9
23:04:33: updatePubData - complete
23:05:28: EliteDangerous window reposition: {X=0,Y=0,Width=0,Height=0} => {X=0,Y=4,Width=1920,Height=1080}
23:05:28: Program.repositionPlotters: 0, rect{X=0,Y=4,Width=1920,Height=1080}
23:05:34: New journal file created: Journal.2024-01-07T230534.01.log
23:05:34: Main.newGame: has old game:False 
23:05:34: Game .ctor
23:05:34: Reading: Journal.2024-01-07T230534.01.log
23:05:34: Game.initializeFromJournal: BEGIN  (FID:), journals.Count:0
23:05:34: Game.initializeFromJournal: END Commander:, starSystem:, systemLocation:, systemBody:, journals.Count:0
23:05:34: Mode change NoFocus => MainMenu
23:05:34: Cmdr loaded: False
23:05:34: migratedScannedOrganicsInEntryId: , migratedNonSystemDataOrganics: 
23:05:42: Game.event => Fileheader
23:05:59: Game.event => Commander
23:06:00: Game.event => LoadGame
23:06:00: Game.initializeFromJournal: BEGIN  (FID:), journals.Count:3
23:06:00: Loaded data from: C:\Users\User\AppData\Roaming\SrvSurvey\SrvSurvey\1.1.0.0\F5439629-live.json
23:06:00: Game.initSystemData: rewind to last FSDJump or CarrierJump
23:06:00: Reading: Journal.2024-01-06T231515.01.log
23:06:00: Game.initSystemData: found last FSDJump, to 'Great Annihilator' (2587943)
23:06:00: Loading SystemData for: 'Great Annihilator' (2587943)
23:06:00: Loaded data from: C:\Users\User\AppData\Roaming\SrvSurvey\SrvSurvey\1.1.0.0\systems\F5439629\Great Annihilator_2587943.json
23:06:00: walkDeep: count: 2
23:06:00: Game.initSystemData: Processing 1 journal items forwards...
23:06:00: Game.initSystemData: playForwards 'Disembark' (6-1-2024 23:01:09)
23:06:00: Game.initSystemData: complete 'Great Annihilator' (2587943), bodyCount: 8
23:06:00: Game.initCargoData: rewind to last Cargo
23:06:00: Game.initCargoData: Processing 0 cargo journal items forwards...
23:06:00: Exception processing event 'LoadGame':
System.ArgumentNullException: Value cannot be null. (Parameter 'values')
   at System.String.JoinCore[T](ReadOnlySpan`1 separator, IEnumerable`1 values)
   at System.String.Join[T](String separator, IEnumerable`1 values)
   at SrvSurvey.game.Game.initCargoData() in D:\a\SrvSurvey\SrvSurvey\SrvSurvey\game\Game.cs:line 669
   at SrvSurvey.game.Game.initializeFromJournal(LoadGame loadEntry) in D:\a\SrvSurvey\SrvSurvey\SrvSurvey\game\Game.cs:line 486
   at System.Dynamic.UpdateDelegates.UpdateAndExecuteVoid2[T0,T1](CallSite site, T0 arg0, T1 arg1)
   at SrvSurvey.game.Game.Journals_onJournalEntry(JournalEntry entry, Int32 index) in D:\a\SrvSurvey\SrvSurvey\SrvSurvey\game\Game.cs:line 708
23:06:16: Mode change MainMenu => Docked
23:06:46: Mode change Docked => CommsPanel
23:07:09: Mode change CommsPanel => Docked
23:08:37: Mode change Docked => GalaxyMap
23:09:39: Mode change GalaxyMap => Docked
23:09:54: Mode change Docked => Flying
23:10:55: Mode change Flying => SuperCruising
23:11:21: Mode change SuperCruising => FSS
23:11:21: Program.Creating plotter: PlotFSS
23:11:21: Program.Showing plotter: PlotFSS
23:13:59: Mode change FSS => SuperCruising
23:13:59: Program.Closing plotter: PlotFSS
23:14:18: Opening link: (length: 45)
https://github.com/njthomson/SrvSurvey/issues

23:14:28: Opening link: (length: 1031)
https://github.com:443/njthomson/SrvSurvey/issues/new?template=crash-report.yml&title=ArgumentNullException+%22Value+cannot+be+null.+(Parameter+%27values%27)%22+at+7-1-2024+23%3A14%3A28+%2B01%3A00&wha

```

**Comment by njthomson** [comment]
@Havym - can you try again with this build?

https://github.com/njthomson/SrvSurvey/actions/runs/7437160359

**Comment by njthomson** [comment]
@Havym - this has been fixed with build `1.1.0.6`
https://github.com/njthomson/SrvSurvey/releases/tag/1.1.0.6
